cloud/scope/vpc_cluster.go

```diff
@@ -686,3 +742,321 @@ func (s *VPCClusterScope) buildCOSObjectHRef() (*string, error) {
+						return false, nil
+					}
+				}
+			} else if !isControlPlane && s.NetworkStatus().WorkerSubnets != nil {
+				for _, wSubnet := range s.NetworkStatus().WorkerSubnets {
+					if *subnet.ID == wSubnet.ID && wSubnet.Ready {
+						s.V(3).Info("found subnet is ready", "subnetID", subnet.ID)
+						return false, nil
+					}
+				}
+			}
+		}
+
+		// Otherwise, if we have a subnet ID, attempt lookup to confirm it exists
+		options := &vpcv1.GetSubnetOptions{
+			ID: subnet.ID,
+		}
+		subnetDetails, _, err := s.VPCClient.GetSubnet(options)
+		if err != nil {
+			return false, fmt.Errorf("error trying to find subnet by id: %s", *subnet.ID)
+		}
+		// If a subnet ID is provided, we assume it must exist, we will not attempt to create a new subnet.
+		if subnetDetails == nil {
+			return false, fmt.Errorf("error failed to find subnet with id %s", *subnet.ID)
+		}
+		s.V(3).Info("Found Subnet with provided id", "subnetID", subnet.ID)
+
+		// Check and update the subnet Network Status based on returned details.
+		return s.checkSubnetStatus(subnetDetails, isControlPlane)
+	} else {
+		// TODO(cjschaef): No name or ID should results in an error currently. We may wish to revisit cases where Subnets already exist, and we don't care to reconcile them (verify they exist).
+		return false, fmt.Errorf("error no subnet name or id defined")
+	}
+
+	// If we have reached this point, we assume the sunbet does not exist yet and we need to create it.
+	s.V(3).Info("creating subnet", "subnetName", subnet.Name)
+	err := s.createSubnet(subnet, isControlPlane)
+	if err != nil {
+		return false, err
+	}
+	s.V(3).Info("Successfully created subnet", "subnetName", subnet.Name)
+
+	// Recommend we requeue reconciliation after subnet was successfully created
+	return true, nil
+}
+
+// buildSubnetsForZones will create a set of Subnets, using default names, for each availability zone within a Region. This is typically used when no subnets were provided, so a set of default subnets gets created.
+func (s *VPCClusterScope) buildSubnetsForZones() ([]infrav1beta2.Subnet, error) {
+	subnets := make([]infrav1beta2.Subnet, 0)
+	zones, err := s.VPCClient.GetVPCZonesByRegion(s.IBMVPCCluster.Spec.Region)
+	if err != nil {
+		return subnets, fmt.Errorf("error unknown failure retrieving zones for region %s: %w", s.IBMVPCCluster.Spec.Region, err)
+	}
+	if len(zones) == 0 {
+		return subnets, fmt.Errorf("error retrieving subnet zones, no zones found in %s", s.IBMVPCCluster.Spec.Region)
+	}
+	for _, zone := range zones {
+		name := fmt.Sprintf("%s-%s", *s.GetServiceName(infrav1beta2.ResourceTypeSubnet), zone)
+		subnets = append(subnets, infrav1beta2.Subnet{
+			Name: ptr.To(name),
+			Zone: ptr.To(zone),
+		})
+	}
+	return subnets, nil
+}
+
+// checkSubnetStatus will check the status of a IBM Cloud Subnet and update the Network Status.
+func (s *VPCClusterScope) checkSubnetStatus(subnetDetails *vpcv1.Subnet, isControlPlane bool) (bool, error) {
```

Comment:
I feel we need to rename this if you are going to update the status also as part of this func.

Reply:
I will change to `update`.